sys/kern/file_syscalls.c

```diff
@@ -33,6 +33,13 @@ int do_write(proc_t *p, int fd, uio_t *uio) {
 
   uio->uio_ioflags |= f->f_flags & IO_MASK;
   error = f->f_ops->fo_write(f, uio);
+  if (error) {
+    if (error == EPIPE) {
+      mtx_lock(&p->p_lock);
+      sig_kill(p, &DEF_KSI_RAW(SIGPIPE));
+      mtx_unlock(&p->p_lock);
+    }
+  }
```

Comment:
It's not a new test.
I suggest to do one change in one pr.

Reply:
That's ok. The change is so small I decided not to burden @hadarai with creating a new PR.